### DISCUSSION TITLE
Use props to pass parameters in the nuxt framework for problems encountered in bg-[]

### DISCUSSION THREAD

**Discussion by fishcy** [tailwindlabs/tailwindcss#11349] [Help]
After the parent component passes the image url to the props props, setting the background image using the bg-[] custom method does not take effect after rendering. My personal guess is that the bottom layer of tailwindcss didn't recognize the props class when packing, so it didn't convert it to the appropriate css. The underlying logic of tailwindcss is not recognized because props have not been replaced, probably due to vue's render function not executing. Is there a good solution?

**Comment by mybearworld** [comment]
Are you doing something like this?

```vue
<script lang="ts" setup>
defineProps<{
  color: string;
}>();
</script>

<template>
  <button :class="[`bg-[${color}]`]">Lorem ipsum dolor sit amet</button>
</template>
```

If so, then that intentionally doesn't work in order to keep Tailwind simple. Tailwind requires all classes be written out in plain text. You'd have to rewrite it to something like this:

```vue
<script lang="ts" setup>
const styles = {
  red: "bg-red-500",
  yellow: "bg-yellow-600",
  green: "bg-lime-400",
};

defineProps<{
  color: keyof typeof styles;
}>();
</script>

<template>
  <button :class="[styles[color]]">Lorem ipsum dolor sit amet</button>
</template>
```

This is a better approach anways because it forces the button to fit in with your design system. If you really need the color to be completely customizable, just use inline styles:

```vue
<script lang="ts" setup>
defineProps<{
  color: string;
}>();
</script>

<template>
  <button :style="{ backgroundColor: color }">
    Lorem ipsum dolor sit amet
  </button>
</template>
```

You can read more [here](https://tailwindcss.com/docs/content-configuration#class-detection-in-depth).

**Reply by fishcy** [reply]
ok, thanks
